src/session/session_dhandle.c

```diff
@@ -404,13 +404,33 @@ static int
 __session_find_shared_dhandle(WT_SESSION_IMPL *session, const char *uri, const char *checkpoint)
 {
     WT_DECL_RET;
+    bool is_checkpoint;
 
-    WT_WITH_HANDLE_LIST_READ_LOCK(session,
-      if ((ret = __wt_conn_dhandle_find(session, uri, checkpoint)) == 0)
-        WT_DHANDLE_ACQUIRE(session->dhandle));
-
-    if (ret != WT_NOTFOUND)
-        return (ret);
+    /*
+     * If we get called as part of the checkpoint, and the handle is the checkpoint to be created,
+     * it is guaranteed that the handle won't exist in the shared dhandle list. Skip searching
+     * through the list, and directly insert into it.
```

Comment:
I think the use of "handle" is confusing since there is no local variable called handle and what you're really referring to is the checkpoint local variable not even the uri. I suggest rewording the first line as:
If this session is taking a checkpoint and is looking for the system checkpoint name to be created, it is guaranteed...